var/spack/repos/builtin/packages/metis/package.py

```diff
@@ -98,12 +98,32 @@ def setup_build_environment(self, env):
 
 class MakefileBuilder(spack.build_systems.makefile.MakefileBuilder, SetupEnvironment):
     @property
-    def build_targets(self):
+    def compile_options(self):
         options = []
         if "+shared" in self.spec:
-            options.append("COPTIONS={0}".format(self.pkg.compiler.cc_pic_flag))
+            options.append(self.pkg.compiler.cc_pic_flag)
+        if self.spec.satisfies("%cce@17:"):
+            options.append("-std=c89")
```

Comment:
(question) do older versions of `%cce` need a different `-std`, or are you adding this bound because you haven't used `%cce:16`?

Reply:
I think this is only needed with the LLVM-based `cce` versions and only versions that use LLVM >= X, where X is 15 or 14, I'm not exactly sure. It's the LLVM version that made implicit function declarations in C errors, if I remember correctly.

Reply:
Thanks! It sounds like this could be expanded, but in this case IMO it's not worth doing so until someone has a chance to try with those versions.